lib/staging/tls/openssl-patch.md

```diff
@@ -0,0 +1,66 @@
+# OpenSSL 3.0.8 patch
+
+The file `openssl-3.0.8-feat-updates-to-support-status_request_v2.patch` is a patch to OpenSSL 3.0.8 to support the `status_request_v2` TLS extension defined in [RFC 6961](https://datatracker.ietf.org/doc/html/rfc6961).
```

Comment:
I have a question here, how is this linked to the openssl library? Is it a local copy that this module is linked against? Does that mean that this module will use version 3.0.8 of openssl while other parts of the project will use another openssl version? Or do we have some mechanism in place in order to make sure that everyone uses this 3.0.8 version?

Reply:
If someone wants to deploy EVerest with support for status_request_v2 then a patched OpenSSL will be required.

My expectation is that EVerest and other code will be built via Yocto to produce the image for deploying to a charging station. Yocto has support for applying patches as part of the build process and hence a single patched version of OpenSSL would be used.

Where Yocto isn't used then my recommendation would be to clone OpenSSL, apply the patch and install (e.g. to /usr/local) and use pkg config variables so that when EVerest is built the /usr/local version of OpenSSL is used.
I would not recommend having some parts of EVerest using different OpenSSL versions.

The main aim is that 15118 TLS connections will work on patched or unpatched versions of OpenSSL.
Unpatched will not support the TLS status_request_v2 handshake extension.

Note that status_request_v2 is only supported for TLS 1.2, TLS 1.3 sessions ignore the status_request_v2 extension.

In summary: the recommended mechanism to use a single patched OpenSSL version is to use Yocto - i.e. meta-everest

Reply:
Understood, but yes, this will make testing a bit difficult. Maybe some extra steps might be required for explanation in the readme in case someone requires this?